File: task1/task1.py
import  sys

def findPath(n, m):
    path = [1]
    currnum = 1
    while True:
        for i in range(1, m):
            if currnum < n:
                currnum += 1
            else:
                currnum = 1
        if currnum == 1:
            break
        path.append(currnum)
    return path

def main():
    if len(sys.argv) != 3:
        print("Enter n and m !")
        sys.exit(1)

    result = findPath(int(sys.argv[1]), int(sys.argv[2]))
    for num in result:
        print(num, end="")

if __name__ == "__main__":
    main()

File: task1/task1.cs
using System;
using System.Collections.Generic;
using System.IO;

class MainClass
{
    public static List<int> findPath(int n, int m)
    {
        List<int> path = new List<int>();
        path.Add(1); int currnum = 1;
        do
        {
            for (int i = 1; i < m; i++)
            {
                if (currnum < n) currnum++;
                else currnum = 1;
            }
            if (currnum == 1) break;
            path.Add(currnum);
        } while (true);

        return path;
    }

    public static void Main(string[] args)
    {
        //TASK1
        if (args.Length != 2) {
            Console.WriteLine("Enter n and m !");
            Environment.Exit(1);
        }

        List<int> result = findPath(int.Parse(args[0]), int.Parse(args[1]));

        foreach (int num in result) {
            Console.Write(num);
        }
        Console.WriteLine();
        
    }
}